validation.go

```diff
@@ -193,27 +191,33 @@ func convertValidations(validations []Validation) []cpluginv1.ParameterValidatio
 	return out
 }
 
-// ApplyConfigValidations a utility function that applies all the validations for parameters, return an error that
+// private struct to group all validation functions
+type validator map[string]Parameter
+
+// applyConfigValidations is a utility function that applies all the validations for parameters, returns an error that
 // consists of a combination of errors from the configurations.
-func applyConfigValidations(params map[string]Parameter, config map[string]string) error {
+func (v validator) applyConfigValidations(config map[string]string) error {
 	// cfg is the config map with default values assigned
-	cfg, multiErr := assignParamDefaults(params, config)
+	cfg, multiErr := v.initConfig(config)
 
-	for pKey, param := range params {
-		if len(strings.TrimSpace(cfg[pKey])) != 0 {
-			multiErr = multierr.Append(multiErr, validateParamType(param, pKey, cfg[pKey]))
+	var err error
+	for pKey := range v {
+		err = v.validateParamType(pKey, cfg[pKey])
+		multiErr = multierr.Append(multiErr, err)
+		// validate param value only if param type is valid
+		if err == nil {
+			multiErr = multierr.Append(multiErr, v.validateParamValue(pKey, cfg[pKey]))
 		}
```

Comment:
Just a nitpick - we should strive to keep the happy path to the left and indent error handling, if we follow this principle the code will be consistent and better readable. So here I would rather turn it around.
```suggestion
		err = v.validateParamType(pKey, cfg[pKey])
		if err != nil {
			// append error and continue with next parameter
			multiErr = multierr.Append(multiErr, err)
			continue
		}
		multiErr = multierr.Append(multiErr, v.validateParamValue(pKey, cfg[pKey]))
```

Reply:
writing the statement `if err==nil` honestly didn't feel right.. so yes I like this more!